### PR TITLE
[SPARK-16964][SQL] Remove private[sql] and private[spark] from sql.execution package

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

This package is meant to be internal, and as a result it does not make sense to mark things as private[sql] or private[spark]. It simply makes debugging harder when Spark developers need to inspect the plans at runtime.

This patch removes all private[sql] and private[spark] visibility modifiers in org.apache.spark.sql.execution.
## How was this patch tested?

N/A - just visibility changes.

### CODE REVIEW

**Comment by rxin** [comment]
cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ class SparkPlanInfo(
   }
 }
 
-private[sql] object SparkPlanInfo {
+private[execution] object SparkPlanInfo {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala]: why keep this private?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/udaf.scala [hunk 2]
```diff
@@ -204,7 +204,7 @@ sealed trait BufferSetterGetterUtils {
 /**
  * A Mutable [[Row]] representing a mutable aggregation buffer.
  */
-private[sql] class MutableAggregationBufferImpl (
+private[aggregate] class MutableAggregationBufferImpl(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/udaf.scala]: here too

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/udaf.scala [hunk 3]
```diff
@@ -266,7 +266,7 @@ private[sql] class MutableAggregationBufferImpl (
 /**
  * A [[Row]] representing an immutable aggregation buffer.
  */
-private[sql] class InputAggregationBuffer private[sql] (
+private[aggregate] class InputAggregationBuffer(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/udaf.scala]: here too

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala [hunk 4]
```diff
@@ -41,22 +42,23 @@ object PartitionDirectory {
...
 
-private[sql] object PartitioningUtils {
+object PartitioningUtils {
   // This duplicates default value of Hive `ConfVars.DEFAULTPARTITIONNAME`, since sql/core doesn't
   // depend on Hive.
-  private[sql] val DEFAULT_PARTITION_NAME = "__HIVE_DEFAULT_PARTITION__"
+  val DEFAULT_PARTITION_NAME = "__HIVE_DEFAULT_PARTITION__"
 
-  private[sql] case class PartitionValues(columnNames: Seq[String], literals: Seq[Literal]) {
+  private[datasources] case class PartitionValues(columnNames: Seq[String], literals: Seq[Literal])
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala]: here too

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala (hunk 1 above)
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala]: If it is something internal to the implementation (i.e. not part of some public field for expressions or commands or query plan), I'm keeping them private for tighten visibility.

**Comment by cloud-fan** [comment]
LGTM, merging to master!

this conflicts with 2.0, so I didn't backport

**Pull request closed without merging**